### PR TITLE
Give an absent float_bar_show_reset_inline the default, not false

### PR DESCRIPTION
**Pull request by btsouth**
## Summary

Stacked on #224 — this is the real bug CodeRabbit flagged there, split out since #224 is scoped as tests-only with no behavior change.

`Settings::default()` sets `float_bar_show_reset_inline` to `true`, but a `settings.json` without the key loaded it as `false`. A fresh install showed the inline reset button on the float bar; every upgraded install hid it, on the same build, with no UI explanation for the difference.

## Cause

`RawSettings` annotated the field with a bare `#[serde(default)]`. That uses the field type's default (`false`) rather than the container's, which `impl Default for RawSettings` seeds correctly from `Settings::default()`. Dropping the attribute lets the container default apply. An explicit `"float_bar_show_reset_inline": false` still loads as `false` — the container default only fills a genuinely absent key.

Also strengthens the empty-config test to require an empty config match `Settings::default()` on every field except `float_bar_contrast` (a deliberate divergence: absence means "fall back to the legacy dark_text flag"), rather than checking fields by hand.

## Verification

Restoring the attribute fails both the new test and the strengthened empty-config test.

## Test plan

- [x] `cargo test --manifest-path rust/Cargo.toml --lib`
- [x] `cargo fmt -- --check`
- [x] `cargo clippy --all-targets -- -D warnings`

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed settings loading so the inline reset option defaults to enabled when omitted.
  * Preserved an explicitly disabled inline reset option during settings serialization and loading.

* **Tests**
  * Added regression coverage for default and explicitly disabled setting values.

### CODE REVIEW

File: rust/src/settings/raw.rs [hunk 1]
```diff
@@ -163,7 +163,11 @@ pub(super) struct RawSettings {
     taskbar_account_by_provider: std::collections::HashMap<String, String>,
     #[serde(default)]
     float_bar_dark_text: bool,
-    #[serde(default)]
+    // NO field-level `#[serde(default)]` here, deliberately. This field defaults
```
[the remark this answers is not in the document]
Comment (btsouth) [reply rust/src/settings/raw.rs]: Real, and I looked into healing it. Not doing so, deliberately.

The setting has a user-facing toggle (`floatbar/SettingsSection.tsx:239`), so `false` on disk has two indistinguishable origins: written by the bug, or written by someone who unticked the box. Under the bug an upgraded user saw the box *unchecked*, which is a coherent state they may well have accepted or actively chosen.

Any heal would have to overwrite every stored `false`, which silently reverses genuine opt-outs. Trading a cosmetic wrong default for overriding an explicit user choice is the worse side of that trade, and it is the same class of defect this PR exists to fix: the app deciding it knows better than the stored value.

The residual cost is bounded: one visible checkbox, one click, no data loss. A version-gated one-shot heal (like `notification_policy_version`) would be the mechanism if this were higher stakes, but it is not worth spending a policy-version bump and the opt-out reversals on an inline reset label.

File: rust/src/settings/raw.rs [hunk 2]
```diff
@@ -163,7 +163,11 @@ pub(super) struct RawSettings {
     taskbar_account_by_provider: std::collections::HashMap<String, String>,
     #[serde(default)]
     float_bar_dark_text: bool,
-    #[serde(default)]
+    // Must stay `default_true`, not a bare `#[serde(default)]`. A bare attribute
+    // uses the FIELD TYPE's default (`false`) rather than the container's, which
+    // is what previously made a fresh install show the inline reset while every
+    // upgraded install hid it.
```
[the remark this answers is not in the document]
Comment (btsouth) [reply rust/src/settings/raw.rs:169]: Accurate about the behavior, but it is not specific to this field and not something this change introduced. I probed it:

```
float_bar_show_reset_inline -> ERR: invalid type: null, expected a boolean
float_bar_show_cost         -> ERR: invalid type: null, expected a boolean
float_bar_dark_text         -> ERR: invalid type: null, expected a boolean
start_minimized             -> ERR: invalid type: null, expected a boolean
sound_enabled               -> ERR: invalid type: null, expected a boolean
refresh_interval_secs       -> ERR: invalid type: null, expected u64
capacity_event_notifications_enabled -> ERR: invalid type: null, expected a boolean
```

An explicit null aborts the load for every non-Option field in RawSettings, roughly 50 of them, identically on main. `capacity_event_notifications_enabled` already used `default_true` before this PR and behaves the same, so the attribute is not the cause. `#[serde(default = ...)]` covers a *missing* key; it never covers a present null.

The `float_bar_contrast` comparison does not carry over either: that field is `Option<String>`, which is exactly why it accepts null. It is not evidence that anything writes null for bools.

So the real issue is repo-wide null tolerance on config load, worth its own change if wanted. Fixing it for this one field would leave the surrounding 50 inconsistent and imply a robustness that is not there. Out of scope for a two-line default fix.
